src/Workspaces/CSharp/Portable/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -122,19 +123,61 @@ public static bool CanRemoveParentheses(this ParenthesizedExpressionSyntax node)
                 : false;
         }
 
+        [ThreadStatic]
+        private static Stack<SyntaxNode> s_nodeStack;
```

Comment:
Why does this need to be a field - I don't see it's state shared across calls to `RemovalMayIntroduceInterpolationAmbiguity`.    Can't it just be a local?